### PR TITLE
Defines interface for mutual TLS

### PR DESCRIPTION
**Pull request by gruyaume**
### Overview

Some charms require knowledge of other charms public TLS certificates for trusted communication with them. This relation interface describes the expected behaviour of any charm claiming to be able to authenticate other charms through TLS communication.

Following this design, I also created a draft for the mutual-tls interface library [here](https://github.com/canonical/mutual-tls-interface/pull/1).

### CODE REVIEW

File: interfaces/mutual_tls/v0/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Both the Requirer and the Provider need to adhere to criteria to be considered compatible with the interface.
+
+### Provider
+
+- Is expected to provide its own public certificate, its CA certificate and its CA chain.
+
+### Requirer
+
+- Is expected to use the provider's certificate and chain to authenticate the provider when communicating with it.
```
Comment (marcoppenheimer) [inline interfaces/mutual_tls/v0/README.md:28]: **question:** Do we want to make all of `ca/certificate/chain` optional? 

I don't think it's necessary to establish a mutual TLS connection to have all 3, and I can imagine scenarios where we might not want to share one or the other, or might not have it.

File: interfaces/mutual_tls/v0/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# `mutual_tls`
+
+## Usage
+
+Some charms require knowledge of other charms public TLS certificates for trusted communication with them. This relation interface describes the expected behaviour of any charm claiming to be able to authenticate other charms through TLS communication.
```
Comment (gruyaume) [inline interfaces/mutual_tls/v0/README.md:5]:
```suggestion
Some charms require knowledge of other charms public TLS certificates for trusted communication with them. This relation interface describes the expected behaviour of any charm claiming to be able to authenticate (or be authenticated) through TLS communication.
```

File: interfaces/mutual_tls/v0/README.md (hunk 1 above)
Comment (gruyaume) [reply interfaces/mutual_tls/v0/README.md:28]: Right now, only the certificate is marked as "required" and the ca and chain are optional.I can change it for all 3 fields to be optional.

Comment (gruyaume) [reply interfaces/mutual_tls/v0/README.md]: Done

**Review by deusebio — APPROVED** [review]
LGTM!

File: interfaces/mutual_tls/v0/schemas/provider.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    "certificate": {
+      "$id": "#/properties/certificate",
+      "type": "string",
+      "title": "Public TLS certificate",
+      "description": "Public TLS certificate"
+    },
+    "ca": {
+      "$id": "#/properties/ca",
+      "type": "string",
+      "title": "CA public TLS certificate",
```
Comment (jnsgruk) [inline interfaces/mutual_tls/v0/schemas/provider.json:28]:
```suggestion
      "title": "CA Public TLS certificate",
```

You're using titlecase elsewhere

Comment (jnsgruk) [reply interfaces/mutual_tls/v0/schemas/provider.json:28]: Actually looking down this is a bit inconsistent throughout.

**Review by jnsgruk — APPROVED** [review]
Content wise this all looks good to me. I've been a total pedant and highlighted some inconsistencies in grammar but otherwise good to merge

File: interfaces/mutual_tls/v0/schemas/provider.json (hunk 3 above)
Comment (gruyaume) [reply interfaces/mutual_tls/v0/schemas/provider.json:28]: The intent was to use capital `P` when it was the first word of the sentence but I see I didn't follow that rule (line 29). I'll fix that up.

File: README.md [hunk 4]
```diff
@@ -25,6 +25,7 @@ To quickly get started, you can copy the `interfaces/__template__` folder to cre
 |               | [`ingress_per_unit`](interfaces/ingress_per_unit/v0/README.md)               | ![Status: Live](https://img.shields.io/badge/Status-Live-darkgreen) |
 | Security      | [`tls_certificates/v0`](interfaces/tls_certificates/v0/README.md)            | ![Status: Live](https://img.shields.io/badge/Status-Live-darkgreen) |
 |               | [`tls_certificates/v1`](interfaces/tls_certificates/v1/README.md)            | ![Status: Draft](https://img.shields.io/badge/Status-Draft-orange)  |
+|               | [`mutual_tls`](interfaces/mutual_tls/v0/README.md)                           | ![Status: Draft](https://img.shields.io/badge/Status-Draft-orange)  |
```
[the remark this answers is not in the document]
Comment (gruyaume) [reply README.md]: Fixed

File: interfaces/mutual_tls/v0/schemas/provider.json (hunk 3 above)
Comment (gruyaume) [reply interfaces/mutual_tls/v0/schemas/provider.json:28]: Fixed, now only uses capital `Public` at beginning of sentences.

**Pull request merged**